drainer/loopbacksync/loopbacksync.go

```diff
@@ -36,36 +37,58 @@ const (
 	ChannelInfo = "channel_info"
 )
 
-// CreateMarkTableDDL is the DDL to create the mark table.
-var CreateMarkTableDDL string = fmt.Sprintf("CREATE TABLE If Not Exists %s (%s bigint not null,%s bigint not null DEFAULT 0, %s bigint DEFAULT 0, %s varchar(64) ,PRIMARY KEY (%s,%s));", MarkTableName, ID, ChannelID, Val, ChannelInfo, ID, ChannelID)
-
-// CreateMarkDBDDL is DDL to create the database of mark table.
-var CreateMarkDBDDL = "create database IF NOT EXISTS retl;"
-
 //LoopBackSync loopback sync info
 type LoopBackSync struct {
 	ChannelID       int64
 	LoopbackControl bool
+	MarkDBName      string
+	MarkTableName   string
 	SyncDDL         bool
+	Index           int64
+	PluginPath      string
+	PluginNames     []string
+	Hooks           []*plugin.EventHooks
+	SupportPlugin   bool
+	RecordID        int
 }
 
 //NewLoopBackSyncInfo return LoopBackSyncInfo objec
-func NewLoopBackSyncInfo(ChannelID int64, LoopbackControl, SyncDDL bool) *LoopBackSync {
+func NewLoopBackSyncInfo(ChannelID int64, LoopbackControl, SyncDDL bool, path string, names []string, SupportPlug bool, mdbname, mtablename string) *LoopBackSync {
 	l := &LoopBackSync{
 		ChannelID:       ChannelID,
 		LoopbackControl: LoopbackControl,
 		SyncDDL:         SyncDDL,
+		Index:           0,
+		PluginPath:      path,
+		PluginNames:     names,
+		SupportPlugin:   SupportPlug,
+		MarkDBName:      strings.TrimSpace(mdbname),
+		MarkTableName:   strings.TrimSpace(mtablename),
+	}
+	if l.SupportPlugin {
+		l.Hooks = make([]*plugin.EventHooks, 4)
+		l.Hooks[plugin.SyncerFilter] = &plugin.EventHooks{}
+
+		l.Hooks[plugin.ExecutorExtend] = &plugin.EventHooks{}
+		l.Hooks[plugin.LoaderInit] = &plugin.EventHooks{}
+		l.Hooks[plugin.LoaderDestroy] = &plugin.EventHooks{}
+
 	}
 	return l
 }
 
 // CreateMarkTable create the db and table if need.
-func CreateMarkTable(db *sql.DB) error {
-	_, err := db.Exec(CreateMarkDBDDL)
+func CreateMarkTable(db *sql.DB, mdbname, mtablename string) error {
+	// CreateMarkDBDDL is DDL to create the database of mark table.
+	var err error
+	var CreateMarkDBDDL = fmt.Sprintf("create database IF NOT EXISTS %s;", mdbname)
```

Comment:
how about use `createMarkDBDDL` because it is a variable in function now?

Reply:
I had fixed it.